.github/workflows/main.yml

```diff
@@ -9,7 +9,7 @@ jobs:
   docker-push:
       runs-on: ubuntu-22.04
       steps:
-        - uses: actions/checkout@v2
+        - uses: actions/checkout@v3
```

Comment:
You can test this by changing the `on.push.branches` to point to this branch. After the run, you can revert it.

Reply:
I redirected to branch `fix-deprecated-warning`. All workflows on `Main, PR and CodeQL` pass succesfully with no deprecated warning. `on.push.branches` has now been reverted to point back to `Main` branch.